xla/stream_executor/cuda/cuda_executor.cc

```diff
@@ -866,8 +911,24 @@ GpuExecutor::CreateDeviceDescription(int device_ordinal) {
     builder.set_pci_bus_id(pci_bus_id);
 
     // Read the NUMA node corresponding to the PCI bus ID out of sysfs.
-    int numa_node = TryToReadNumaNode(pci_bus_id, device_ordinal);
-    builder.set_numa_node(numa_node);
+    std::optional<int> numa_node =
+        TryToReadNumaNode(pci_bus_id, device_ordinal);
+    if (numa_node.has_value()) {
+      if (*numa_node < 0) {  // See http://b/18228951 for details on this path.
```

Comment:
Can a Googler check if this bug is still relevant? This "fudge failure to a valid value" logic makes the test logic a bit cumbersome.

Reply:
This is a bug from 2014 and I think most of the involved people are no longer at google

Reply:
-1 for numa_node is still relevant, it's a kernel definition for "no numa node"

Reply:
Right, we should certainly handle that gracefully (it's even the case that the CI runners hit). My intent with this PR is that if you don't know the numa node, you just allocate without pinning.

The thing that's a little awkward is that -1 gets massaged into a 0 in the device description, which is lossy. That shows up in the test logic where you don't know anymore whether you expect to get an unpinned allocation or an allocation pinned to node 0.